ksql-engine/src/test/java/io/confluent/ksql/analyzer/AggregateAnalyzerTest.java

```diff
@@ -14,205 +14,267 @@
-    Assert.assertTrue(aggregateAnalysis.getFunctionList().size() == 4);
-    Assert.assertTrue(aggregateAnalysis.getFunctionList().get(0).getName().getSuffix()
-                          .equalsIgnoreCase("sum"));
-    Assert.assertTrue(aggregateAnalysis.getFunctionList().get(1).getName().getSuffix()
-                          .equalsIgnoreCase("count"));
-    Assert.assertTrue(aggregateAnalysis.getFunctionList().get(2).getName().getSuffix()
-                          .equalsIgnoreCase("sum"));
-
-    Assert.assertTrue(aggregateAnalysis.getAggregateFunctionArguments().size() == 4);
-    Assert.assertTrue(aggregateAnalysis.getAggregateFunctionArguments().get(0).toString()
-                          .equalsIgnoreCase("(TEST1.COL3 * TEST1.COL0)"));
-    Assert.assertTrue(aggregateAnalysis.getAggregateFunctionArguments().get(1).toString()
-                          .equalsIgnoreCase("TEST1.COL1"));
-    Assert.assertTrue(aggregateAnalysis.getAggregateFunctionArguments().get(2).toString()
-                          .equalsIgnoreCase("(FLOOR(TEST1.COL3) * 3.0)"));
-    Assert.assertTrue(aggregateAnalysis.getNonAggResultColumns().get(0).toString()
-                          .equalsIgnoreCase("test1.col1"));
-
-    Assert.assertTrue(aggregateAnalysis.getFinalSelectExpressions().size() == 3);
-
-    Assert.assertTrue(aggregateAnalysis.getRequiredColumnsList().size() == 3);
-    Assert.assertTrue(aggregateAnalysis.getRequiredColumnsList().get(1).toString()
-                          .equalsIgnoreCase("test1.col3"));
-    Assert.assertTrue(aggregateAnalysis.getHavingExpression() instanceof ComparisonExpression);
-    Assert.assertTrue(aggregateAnalysis.getHavingExpression().toString().equalsIgnoreCase(""
-                                                                                          + ""
-                                                                                          + "(KSQL_AGG_VARIABLE_3 > 10)"));
+    analysis = new MutableAggregateAnalysis();
+    analyzer = new AggregateAnalyzer(analysis, DEFAULT_ARGUMENT, functionRegistry);
+  }
+
+  @Test
+  public void shouldCaptureSelectNonAggregateFunctionArguments() {
+    // When:
+    analyzer.processSelect(FUNCTION_CALL);
+
+    // Then:
+    assertThat(analysis.getNonAggregateSelectExpressions().get(FUNCTION_CALL), contains(COL0));
+  }
+
+  @Test
+  public void shouldCaptureSelectDereferencedExpression() {
+    // When:
+    analyzer.processSelect(COL0);
+
+    // Then:
+    assertThat(analysis.getNonAggregateSelectExpressions().get(COL0), contains(COL0));
+  }
+
+  @Test
+  public void shouldCaptureOtherSelectsWithEmptySet() {
+    // Given:
+    final Expression someExpression = mock(Expression.class);
+
+    // When:
+    analyzer.processSelect(someExpression);
+
+    // Then:
+    assertThat(analysis.getNonAggregateSelectExpressions().get(someExpression), is(empty()));
+  }
+
+  @Test
+  public void shouldNotCaptureOtherNonAggregateFunctionArgumentsAsNonAggSelectColunms() {
```

Comment:
```suggestion
  public void shouldNotCaptureOtherNonAggregateFunctionArgumentsAsNonAggSelectColumns() {
```
nit: "Colunms" -> "Columns" typo

Reply:
done.